decayAmplitude/massDependence.h

```diff
@@ -153,14 +156,177 @@ namespace rpwa {
+			: massDependence(),
+			  _mMin(mMin),
+			  _mMax(mMax),
+			  _coefficients(coefficients) { }
+		virtual ~polynomialMassDependence()   { }
+
+		virtual std::complex<double> amp(const isobarDecayVertex&);
+
+		virtual std::string name() const { return "polynomial"; }  ///< returns label used in graph visualization, reporting, and key file
+
+		const std::vector<std::complex<double> >& getCoefficients() const { return _coefficients; }
+		double                                    getMassMin()      const { return _mMin;         }
+		double                                    getMassMax()      const { return _mMax;         }
+
+	private:
+
+		double                             _mMin;          ///< Mass point that gets scaled to -1. // mMin = -1, mMax = 1 -> no scaling
+		double                             _mMax;          ///< Mass point that gets scaled to  1.
+
+		std::vector<std::complex<double> > _coefficients;  ///< Coefficients of the polynomial
+
+	};
+
+
+	typedef boost::shared_ptr<polynomialMassDependence> polynomialMassDependencePtr;
+
+
+	inline
+	polynomialMassDependencePtr
+	createPolynomialMassDependencePtr(const std::vector<std::complex<double> >& coefficients, const double mMin, const double mMax)
+	{
+		polynomialMassDependencePtr massDep(new polynomialMassDependence(coefficients, mMin, mMax));
+		return massDep;
+	}
+
+
+	//////////////////////////////////////////////////////////////////////////////
+	/// Brief complex exponential mass dependence with certain degree
+	class complexExponentialMassDependence : public massDependence {
+
+	public:
+
+		complexExponentialMassDependence(const int degree, const double mMin, const double mMax) : massDependence(), _degree(degree), _mMin(mMin), _mMax(mMax) { }
+		virtual ~complexExponentialMassDependence()                                                                                                            { }
+
+		virtual std::complex<double> amp(const isobarDecayVertex&);
+
+		virtual std::string name() const { return "complexExponential"; }  ///< returns label used in graph visualization, reporting, and key file
+
+		int    getDegree()  const { return _degree; }
+		double getMassMin() const { return _mMin;   }
+		double getMassMax() const { return _mMax;   }
+
+	private:
+
+		int    _degree;  ///< Degree of the expansion
+		double _mMin;    ///< Mass point that gets scaled to 0.
+		double _mMax;    ///< Mass point that gets scaled to 2.
+
+	};
+
+
+	typedef boost::shared_ptr<complexExponentialMassDependence> complexExponentialMassDependencePtr;
+
+
+	inline
+	complexExponentialMassDependencePtr
+	createComplexExponentialMassDependencePtr(const int degree, const double mMin, const double mMax)
+	{
+		complexExponentialMassDependencePtr massDep(new complexExponentialMassDependence(degree, mMin, mMax));
+		return massDep;
+	}
+
+
+	//////////////////////////////////////////////////////////////////////////////
+	/// Brief complex TF1 mass dependence with arbitrary function
```

Comment:
mass dependence described by two arbitrary TF1 functions for real and imaginary part